index.js

```diff
@@ -343,163 +342,160 @@ Filter.prototype._handleFile = function(relativePath, srcDir, destDir, entry, ou
-    let outputPath = filter.getDestFilePath(relativePath, entry);
+      if (outputPath == null) {
+        throw new Error('canProcessFile("' + relativePath +
+                        '") is true, but getDestFilePath("' +
+                        relativePath + '") is null');
+      }
 
-    if (outputPath == null) {
-      throw new Error('canProcessFile("' + relativePath +
-                      '") is true, but getDestFilePath("' +
-                      relativePath + '") is null');
-    }
+      outputPath = destDir + '/' + outputPath;
 
-    outputPath = destDir + '/' + outputPath;
+      if (isChange) {
+        let isSame = fs.readFileSync(outputPath, 'UTF-8') === outputString;
+        if (isSame) {
+          this._logger.debug('[change:%s] but was the same, skipping', relativePath, isSame);
+          return;
+        } else {
+          this._logger.debug('[change:%s] but was NOT the same, writing new file', relativePath);
+        }
+      }
 
-    if (isChange) {
-      let isSame = fs.readFileSync(outputPath, 'UTF-8') === outputString;
-      if (isSame) {
-        this._logger.debug('[change:%s] but was the same, skipping', relativePath, isSame);
-        return;
-      } else {
-        this._logger.debug('[change:%s] but was NOT the same, writing new file', relativePath);
+      try {
+        fs.writeFileSync(outputPath, outputString, {
+          encoding: outputEncoding
+        });
+
+      } catch(e) {
+        // optimistically assume the DIR was patched correctly
+        mkdirp.sync(path.dirname(outputPath));
+        fs.writeFileSync(outputPath, outputString, {
+          encoding: outputEncoding
+        });
       }
-    }
 
-    try {
-      fs.writeFileSync(outputPath, outputString, {
-        encoding: outputEncoding
-      });
+      return outputString;
+    });
+  }
 
-    } catch(e) {
-      // optimistically assume the DIR was patched correctly
-      mkdirp.sync(path.dirname(outputPath));
-      fs.writeFileSync(outputPath, outputString, {
-        encoding: outputEncoding
-      });
-    }
+  /**
+   * @param contents {string}
+   * @param relativePath {string}
+   * @returns {string}
+   */
+  processString (contents, relativePath) { // jshint ignore:line
+    throw new Error(
+        'When subclassing broccoli-persistent-filter you must implement the ' +
+        '`processString()` method.');
+  }
 
-    return outputString;
-  });
-};
+  postProcess (result /*, relativePath */) {
+    return result;
+  }
+}
 
-/**
- * @param contents {string}
- * @param relativePath {string}
- * @returns {string}
- */
-Filter.prototype.processString = function(contents, relativePath) { // jshint ignore:line
-  throw new Error(
-      'When subclassing broccoli-persistent-filter you must implement the ' +
-      '`processString()` method.');
-};
-
-Filter.prototype.postProcess = function(result /*, relativePath */) {
-  return result;
-};
+module.exports = Filter;
+module.exports.shouldPersist = shouldPersist;
```

Comment:
Probably should make this static instead of adding it to `Filter` like this

Reply:
Yes. What this is doing now is _effectively_ the same as:

```js
let SomeValue = Filter;
SomeValue.shouldPersist = shouldPersist;
```

If we actually expect `Filter` to have a `shouldPersist` property on it, then it seems better to make it a static method on the class.

Reply:
I do not have much context around where it should exist. I will check with @chriseppstein I think he will know more about this.

Reply:
I think @stefanpenner has this context... I don't understand why this is exposed in the first place. I'd guess that it's for testing purposes or so a subclass can call it directly without being allowed to override the behavior.

Reply:
I moved it to a static function in Filter class.